src/Cli/dotnet/commands/dotnet-test/TestCommandParser.cs

```diff
@@ -152,38 +155,58 @@ private static CliOption<string> CreateBlameHangDumpOption()
 
         private static readonly CliCommand Command = ConstructCommand();
 
-
-
         public static CliCommand GetCommand()
         {
             return Command;
         }
 
-        private static bool IsTestingPlatformEnabled()
+        private static string GetTestRunnerName()
         {
-            var testingPlatformEnabledEnvironmentVariable = Environment.GetEnvironmentVariable("DOTNET_CLI_TESTINGPLATFORM_ENABLE");
-            var isTestingPlatformEnabled = testingPlatformEnabledEnvironmentVariable == "1" || string.Equals(testingPlatformEnabledEnvironmentVariable, "true", StringComparison.OrdinalIgnoreCase);
-            return isTestingPlatformEnabled;
+            string defaultTestingSdk = CliConstants.VSTest;
+
+            string? globalJsonPath = SdkDirectoryWorkloadManifestProvider.GetGlobalJsonPath(Environment.CurrentDirectory);
+
+            if (!string.IsNullOrEmpty(globalJsonPath))
+            {
+                JsonNode globalJson = JsonObject.Parse(File.ReadAllText(globalJsonPath));
+                JsonNode? testSection = globalJson[CliConstants.TestSectionKey];
+
+                if (testSection is null)
+                {
+                    return defaultTestingSdk;
+                }
+
+                JsonSerializerOptions JsonSerializerOptions = new JsonSerializerOptions
+                {
+                    PropertyNamingPolicy = JsonNamingPolicy.CamelCase,
+                };
+                var testSettings = JsonSerializer.Deserialize<TestSettings>(testSection, JsonSerializerOptions);
+
+                if (testSettings?.Runner?.Name is not null)
+                {
+                    return testSettings.Runner.Name;
+                }
+            }
+            return defaultTestingSdk;
         }
 
         private static CliCommand ConstructCommand()
         {
 #if RELEASE
             return GetVSTestCliCommand();
 #else
-            bool isTestingPlatformEnabled = IsTestingPlatformEnabled();
-            string testingSdkName = isTestingPlatformEnabled ? "testingplatform" : "vstest";
+            string testRunnerName = GetTestRunnerName();
 
-            if (isTestingPlatformEnabled)
+            if (testRunnerName.Equals(CliConstants.VSTest, StringComparison.OrdinalIgnoreCase))
             {
-                return GetTestingPlatformCliCommand();
+                return GetVSTestCliCommand();
             }
-            else
+            else if (testRunnerName.Equals(CliConstants.MicrosoftTestingPlatform, StringComparison.OrdinalIgnoreCase))
             {
-                return GetVSTestCliCommand();
+                return GetTestingPlatformCliCommand();
             }
 
-            throw new InvalidOperationException($"Testing sdk not supported: {testingSdkName}");
+            throw new InvalidOperationException($"Test runner not supported: {testRunnerName}");
```

Comment:
We will have to use resx everywhere later.

Reply:
Yes, we are already doing that. I will fix this.